clients/src/main/java/org/apache/kafka/common/PartitionInfo.java

```diff
@@ -60,12 +69,19 @@ public int partition() {
     }
 
     /**
-     * The node id of the node currently acting as a leader for this partition or null if there is no leader
+     * The node currently acting as a leader for this partition or null if there is no leader
      */
     public Node leader() {
         return leader;
     }
 
+    /**
+     * The epoch of the partition's leader.
```

Comment:
I think it's the "The leader epoch of this partition if known".

Reply:
`Optional` here, right?